### PR TITLE
[13.0][IMP] github_connector_odoo: Code analysis

### PR DESCRIPTION
**Pull request by victoralmau**
Implements Analysis source code in odoo module versions, now add new field called "has_odoo_addons" (boolean) in Analysis rule that allow (if defined) analyze odoo_module_versions and save info about it

In 'Repository Branch' / 'Code Analysis' show the info too related to Analysis rules odoo_module_versions

Locked by:
- [x] https://github.com/OCA/interface-github/pull/63 `github_connector`

Please @pedrobaeza can you review it?

@Tecnativa TT25583

### CODE REVIEW

File: github_connector_odoo/readme/USAGE.rst [hunk 1]
```diff
@@ -42,6 +42,11 @@ manifest file.
 
 .. image:: /github_connector_odoo/static/description/odoo_python_libs.png
 
+**Analysis source code**
+
+Implements Analysis source code in odoo module versions, now add new field called "has_odoo_addons" (boolean) in Analysis rule that allow (if defined) analyze odoo_module_versions and save info about it
+
+In 'Repository Branch' / 'Code Analysis' show the info too related to Analysis rules odoo_module_versions
```
Comment (joao-p-marques) [inline github_connector_odoo/readme/USAGE.rst]:
```suggestion
**Soource code analysis**

Implements Source code analysis odoo module versions, now addiing new field called "has_odoo_addons" (boolean) in Analysis rule that (if defined) allows to analyze odoo_module_versions and save info about it

In 'Repository Branch' / 'Code Analysis', shows the info related to odoo_module_versions rules analysis too.

```

**Review by joao-p-marques — COMMENTED** [review]
Nothing special, just some phrasing changes that might make it easier to understand IMHO.

Code LGTM :+1:

**Comment by victoralmau** [comment]
Ping @pedrobaeza

**Review by joao-p-marques — APPROVED** [review]

**Comment by victoralmau** [comment]
Sorry, it's not ready yet. It's necessary to add test and be sure to work fine analyze action.

**Comment by pedrobaeza** [comment]
Please comment when ready.

**Comment by victoralmau** [comment]
Ready to review it now.

**Review by joao-p-marques — APPROVED** [review]

**Comment by victoralmau** [comment]
IMO it's finally ready to review.

**Comment by pedrobaeza** [comment]
@cesar-tecnativa please review this PR

**Review by cesar-tecnativa — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please fw-port it directly to v15

**Comment by victoralmau** [comment]
> Please fw-port it directly to v15

The module does not yet exist in v15 (PR exists). Do you think it's better to add this change to v14 + merge it and then add it to the PR of v15 migration?

**Comment by pedrobaeza** [comment]
OK, let's include it in v14 as well

**Pull request merged**